### ISSUE TITLE
Trying to get the evaluation metrics but getting error

### ISSUE THREAD

**Issue by AdityaBala05** [ultralytics/ultralytics#18863]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I have trained my Yolov8 multi class classification for 4 classes and validated the test data with an accuracy of 88%

I tried getting Precision and recall using metrics.box() but got an error stating Classify metrics does not have .box

How do I plot Roc curve and find f1 score, precision ,recall for this and are there any other metrics that I can show apart from Accuracy, precision,recall,f1 score,Roc plot

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AdityaBala05, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question or a metrics-related issue, please provide additional details about the command/code you are using and error messages (if any) in full. This will help us accurately understand and address your inquiry. Ensure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for validation and evaluations.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

➡️ This is an automated response, but no worries - an Ultralytics engineer will be here to assist you soon! 😊

**Comment by glenn-jocher** [comment]
@AdityaBala05 thank you for your question! The error occurs because `.box` metrics are specific to object detection tasks, not classification. For multi-class classification in YOLOv8, you can directly compute metrics like Precision, Recall, F1 Score, and ROC curves using standard classification evaluation methods.

To plot ROC curves and calculate these metrics, you can use libraries like `sklearn`. Additionally, you can explore metrics like Confusion Matrix, Precision-Recall curves, and Log Loss to complement your evaluation. For more details on YOLO performance metrics, refer to [YOLO Performance Metrics Documentation](https://docs.ultralytics.com/guides/yolo-performance-metrics/).

**Comment by AdityaBala05** [comment]
> [@AdityaBala05](https://github.com/AdityaBala05) thank you for your question! The error occurs because `.box` metrics are specific to object detection tasks, not classification. For multi-class classification in YOLOv8, you can directly compute metrics like Precision, Recall, F1 Score, and ROC curves using standard classification evaluation methods.
> 
> To plot ROC curves and calculate these metrics, you can use libraries like `sklearn`. Additionally, you can explore metrics like Confusion Matrix, Precision-Recall curves, and Log Loss to complement your evaluation. For more details on YOLO performance metrics, refer to [YOLO Performance Metrics Documentation](https://docs.ultralytics.com/guides/yolo-performance-metrics/).

How do I plot the  Precision-Recall curve and calculate Log Loss

**Comment by Y-T-G** [comment]
They are not calculated for classification models

**Comment by AdityaBala05** [comment]
> [@AdityaBala05](https://github.com/AdityaBala05) thank you for your question! The error occurs because `.box` metrics are specific to object detection tasks, not classification. For multi-class classification in YOLOv8, you can directly compute metrics like Precision, Recall, F1 Score, and ROC curves using standard classification evaluation methods.
> 
> To plot ROC curves and calculate these metrics, you can use libraries like `sklearn`. Additionally, you can explore metrics like Confusion Matrix, Precision-Recall curves, and Log Loss to complement your evaluation. For more details on YOLO performance metrics, refer to [YOLO Performance Metrics Documentation](https://docs.ultralytics.com/guides/yolo-performance-metrics/).




> They are not calculated for classification models

Thank you so much for solving this query, i saw this code earlier but was pondering whether there was any way i can get these values automatically while validation.Sorry to bother you but I have few doubts
1) Do I need to put bounding box on images for plotting ROC curve for image classification in YoloV8, if yes how many images should be having bounding box, my test data has 496 images.
2) If yes or if no for above, how do i plot that curve in my colab
3) Which are the best XAI models for the same and it would be helpful if you share any resources

**Comment by Y-T-G** [comment]
You can use scikit-learn to get the metrics.
